File: src/features/typing/data/go/product_of_array_except_self.go
package main

func productExceptSelf(nums []int) []int {
	n := len(nums)
	result := make([]int, n)
	prefix := 1
	for i, num := range nums {
		result[i] = prefix
		prefix *= num
	}
	suffix := 1
	for i := n - 1; i >= 0; i-- {
		result[i] *= suffix
		suffix *= nums[i]
	}
	return result
}

File: src/features/typing/data/dart/product_of_array_except_self.dart
class Solution {
  List<int> productExceptSelf(List<int> nums) {
    final n = nums.length;
    final result = List<int>.filled(n, 1);
    var prefix = 1;
    for (int i = 0; i < n; i++) {
      result[i] = prefix;
      prefix *= nums[i];
    }
    var suffix = 1;
    for (int i = n - 1; i >= 0; i--) {
      result[i] *= suffix;
      suffix *= nums[i];
    }
    return result;
  }
}
